testsuite/features/step_definitions/navigation_steps.rb

```
@@ -161,6 +161,31 @@
   raise ScriptError, "Checkbox #{identifier} not unchecked." if has_checked_field?(identifier)
 end
 
+When(/^I (check|uncheck) "([^"]*)" by label$/) do |action, identifier|
+  begin
+    checkbox = find(:xpath, "//input[@type='checkbox' and @id = //label[normalize-space(.) = '#{identifier}']/@for]")
+    if action == 'check'
+      action_boolean = true
+    elsif action == 'uncheck'
+      action_boolean = false
+    else
+      raise StandardError, "Variable indicated in step:#{action} not correctly indicated"
+    end
+    checkbox.set(action_boolean)
+  rescue Capybara::ElementNotFound => e
+    puts "ElementNotFound. Error message: #{e.message}"
+  rescue Capybara::Ambiguous => e
+    puts "Element is ambiguous. Error message: #{e.message}"
+  rescue StandardError => e
+    puts "StandardError. Error message: #{e.message}"
```

If we have errors, we don't want to catch them and just print a log.
If we have an error, we want to raise the error, and handle the issue.
Please remove these rescues.

---

The action here is checking or unchecking checkboxes and the error that should be raised is if the checkbox is checked or unchecked respectively, not the ones that I catch in these try-caths. The ones that I catch here are just to catch errors in the finding of the label, something not relevant for the purpouse of this. The one that really matters is raised bellow.

---

But I should use log instead of put to write the errors, that's right

---

Outdated